e2e/lifecycle_hooks/lifecycle_hooks_test.go

```diff
@@ -0,0 +1,72 @@
+package lifecycle_hooks
```

Comment:
Out of curiosity, other than the long bit of boiler plate (I should factor that out into a reusable thing...), what did you think of writing an e2e test?

Reply:
I liked. It was easy to understand and implement it.